demos/common/hfsm/az_iot_hfsm.c

```diff
@@ -0,0 +1,324 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file az_iot_hfsm.c
+ * @brief HFSM for Azure IoT Operations.
+ * 
+ * @details Implements fault handling for Device Provisioning + IoT Hub operations
+ */
+
+#include <stdint.h>
+#include <inttypes.h>
+
+#include "az_hfsm.h"
+#include "az_hfsm_pal_timer.h"
+#include "az_iot_hfsm.h"
+
+// TODO: change logging to az_log_internal.
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Include logging header files and define logging macros in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define the LIBRARY_LOG_NAME and LIBRARY_LOG_LEVEL macros depending on
+ * the logging configuration for DEMO.
+ * 3. Include the header file "logging_stack.h", if logging is enabled for DEMO.
+ */
+
+#include "logging_levels.h"
+
+/* Logging configuration for the Demo. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "IoT_HFSM"
+#endif
+
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_INFO
+#endif
+
+/* 
+ * The function prints to the console before the network is connected;
```

Comment:
This function prints...

Reply:
Logging is actually a todo: I want to move all this to [az_log ](https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/inc/azure/core/az_log.h#L95) instead of taking a hard dependency on FreeRTOS then create a callback to properly print them out at console.

(Everything here is a copy/paste from other places where we use FreeRTOS's logging.)